### PR TITLE
Added "Progress Tracker" component

### PR DESCRIPTION
**Pull request by wtfuii**
Hey there,

I added support for the "Progress Tracker" component.
Let me know if you have any remarks. :)

BR,
Klaus

### CODE REVIEW

File: src/components/progress-tracker/AuiProgressTracker.vue [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    <slot></slot>
+  </ol>
+</template>
+
+<script>
+  export default {
+  }
+</script>
+
+<style scoped>
```
Comment (damienix) [inline src/components/progress-tracker/AuiProgressTracker.vue]: The style section is not necessary here

File: src/components/progress-tracker/AuiProgressTrackerStep.vue [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+<script>
+  export default {
+    props: {
+      active: Boolean
+    }
+  }
+</script>
+
+<style scoped>
```
Comment (damienix) [inline src/components/progress-tracker/AuiProgressTrackerStep.vue]: Also here

File: src/components/progress-tracker/AuiProgressTrackerStep.vue [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<template>
+  <li class="aui-progress-tracker-step" :class="{'aui-progress-tracker-step-current': active}" style="width:25%"><span><slot></slot></span></li>
```
Comment (damienix) [inline src/components/progress-tracker/AuiProgressTrackerStep.vue]: Please break this line as it's pretty long.

There's also a major problem with the styling here. `width: 25%` seems to only be working properly to up to 4 items. Calculating proper width (100% divided by number of items) here is not very convenient here as we'd have to count the steps (not that pleasant). What I found is a good heuristic here is to set width to 1%. As these items are in the table, they'll expand anyway. The only difference in the behaviour is that the columns in this case don't respect their `max-width`, but that's fine to me in this case.

File: src/docs/component/ProgressTracker.vue [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            <aui-progress-tracker-step active>Third Step</aui-progress-tracker-step>
+            <aui-progress-tracker-step>Fourth Step</aui-progress-tracker-step>
+          </aui-progress-tracker>
+
+          <aui-progress-tracker>
+            <aui-progress-tracker-step active><a href="#">First Link</a></aui-progress-tracker-step>
+            <aui-progress-tracker-step><a href="#">Second Link</a></aui-progress-tracker-step>
+            <aui-progress-tracker-step><a href="#">Third Link</a></aui-progress-tracker-step>
+            <aui-progress-tracker-step><a href="#">Fourth Link</a></aui-progress-tracker-step>
+          </aui-progress-tracker>
```
Comment (damienix) [inline src/docs/component/ProgressTracker.vue]: Please add examples for more and less than 4 steps. (you can change one of the existing ones too)

File: src/components/progress-tracker/AuiProgressTrackerStep.vue (hunk 3 above)
Comment (wtfuii) [reply src/components/progress-tracker/AuiProgressTrackerStep.vue]: I already tested it, 25% seemed to be correct for an arbitray number of steps, but I will double check this. Thanks!

Comment (damienix) [reply src/components/progress-tracker/AuiProgressTrackerStep.vue]: I was testing it on Chrome, try with different window widths.

**Comment by wtfuii** [comment]
I implemented the requested changes, though I'm still not 100% happy with the solution for `width` on steps. The constraint here is AUI itself, as its `min-width` on steps causes the content to grow too wide on smaller screens.
I guess it should be fine like this. 
BR,
Klaus

**Comment by damienix** [comment]
Yes, it's just how it works in AUI. Perhaps there is little value in creating some special switch/mechanism for this as you can just create your own simple CSS snippet to clear this `min-width` for the columns if you really need it.

You forgot to update the code snippet :) I'm merging the PR once you fix it.

P.S. I'm releasing version 1.0 of this library tomorrow so this PR will be included there :) i.e. I don't plan to release 0.X with this component. There won't be too many braking changes (mostly aui-spinner) so shouldn't be painful to upgrade.

**Comment by damienix** [comment]
Hi, I just wanna let you know that I've just published 1.0.1 that includes the progress tracker component :)

Cheers,
Damian

**Pull request merged**